src/services/fileServices/MdPageServices/__tests__/PageService.spec.ts

```diff
@@ -0,0 +1,940 @@
+import { CollectionPageService } from "@root/services/fileServices/MdPageServices/CollectionPageService"
+import { ContactUsPageService } from "@root/services/fileServices/MdPageServices/ContactUsPageService"
+import { UnlinkedPageService } from "@root/services/fileServices/MdPageServices/UnlinkedPageService"
+import {
+  CollectionPageName,
+  ContactUsPageName,
+  HomepageName,
+  ResourceCategoryPageName,
+  SubcollectionPageName,
+  UnlinkedPageName,
+} from "@root/types/pages"
+import { Brand } from "@root/types/util"
+
+import { HomepagePageService } from "../HomepagePageService"
+import { PageService } from "../PageService"
+import { ResourcePageService } from "../ResourcePageService"
+import { SubcollectionPageService } from "../SubcollectionPageService"
+
+const mockContactUsService = jest.mocked<ContactUsPageService>(({
+  read: jest.fn(),
+} as unknown) as ContactUsPageService)
+const mockCollectionPageService = jest.mocked<CollectionPageService>(({
+  read: jest.fn(),
+} as unknown) as CollectionPageService)
+const mockSubcollectionPageService = jest.mocked<SubcollectionPageService>(({
+  read: jest.fn(),
+} as unknown) as SubcollectionPageService)
+const mockHomepageService = jest.mocked<HomepagePageService>(({
+  read: jest.fn(),
+} as unknown) as HomepagePageService)
+const mockResourcePageService = jest.mocked<ResourcePageService>(({
+  read: jest.fn(),
+} as unknown) as ResourcePageService)
+const mockUnlinkedPageService = jest.mocked<UnlinkedPageService>(({
+  read: jest.fn(),
+} as unknown) as UnlinkedPageService)
+const mockResourceRoomDirectoryService = jest.mocked<ResourceRoomDirectoryService>(
+  ({
+    getResourceRoomDirectoryName: jest.fn(),
+  } as unknown) as ResourceRoomDirectoryService
+)
+const pageService = new PageService({
+  contactUsService: mockContactUsService,
+  collectionPageService: mockCollectionPageService,
+  subCollectionPageService: mockSubcollectionPageService,
+  homepageService: mockHomepageService,
+  resourcePageService: mockResourcePageService,
+  unlinkedPageService: mockUnlinkedPageService,
+  resourceRoomDirectoryService: mockResourceRoomDirectoryService,
+})
+
+const MOCK_RESOURCE_ROOM_NAME = "resources"
+const MOCK_UNLINKED_PAGE_NAME = "some_page"
+const MOCK_COLLECTION_NAME = "a_collection"
+const MOCK_SUBCOLLECTION_NAME = "submarine"
+const MOCK_RESOURCE_CATEGORY_NAME = "meow"
+const createMockStagingPermalink = (mockPageName: string) =>
+  `${MOCK_STAGING_URL_GITHUB}/${mockPageName}`
+const createMockFrontMatter = (mockPageName: string) => ({
+  fileName: MOCK_UNLINKED_PAGE_NAME,
+  content: {
+    frontMatter: {
+      permalink: `/${mockPageName}`,
+    },
+    pageBody: {},
+  },
+  sha: "",
+})
+
+describe("PageService", () => {
+  describe("parsePageName", () => {
+    it("should only accept 'index.md' as the homepage name", async () => {
+      // Arrange
+      const expected = ok({
+        name: HOMEPAGE_NAME,
+        kind: "Homepage",
```

Comment:
I think this concerns mainly the original PR, but should we change this to use an enum instead?

Reply:
actually i ran into troubles with using enums as types in the past and hence, i didn't use it here. i think the original PR has a comment about this raised by kishore also